packages/editor-ui/src/stores/npsSurvey.store.ts

```diff
@@ -135,19 +139,24 @@ export const useNpsSurveyStore = defineStore('npsSurvey', () => {
 			return;
 		}
 
-		const promptsData: IN8nPrompts = await getPromptsData(
-			settingsStore.settings.instanceId,
-			currentUserId.value ?? '',
-		);
+		try {
+			promptsData.value = await getPromptsData(
+				settingsStore.settings.instanceId,
+				currentUserId.value ?? '',
```

Comment:
Do we even want to fetch the prompts data if `currentUserId` is not set? Why not skip the whole requests then? In what case might this happen? Is there a possible race condition that this is called before `currentUserId` has value?

Reply:
Sure, I can add this. Technically no, since specific user actions are needed to trigger this, for example save a workflow. 
I think this logic predates user management so maybe that's why. We did not have user id, just instance id.

Reply:
for a long time in n8n until v1, setting up an owner user was optional, even after we added user management. so userId could be undefined but a user is logged in.

Reply:
If it should always be defined, it would be nicer to have an `assert` to satisfy TS compiler and to communicate that this should always be defined when this gets called. Otherwise it can cause nasty bugs that are hard to debug (e.g. if there's a bug that this function gets always called before the `currentUserId` is defined the nps survey will never get initialized properly).

Reply:
Yes that makes sense. Thanks for the suggestion